src/components/Button/index.tsx

```diff
@@ -318,8 +275,9 @@ function Button({
                 isDisabled && !danger && !success ? styles.buttonDisabled : undefined,
                 shouldRemoveRightBorderRadius ? styles.noRightBorderRadius : undefined,
                 shouldRemoveLeftBorderRadius ? styles.noLeftBorderRadius : undefined,
-                icon || shouldShowRightIcon ? styles.alignItemsStretch : undefined,
-                ...innerStyles,
+                // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
+                'text' in rest && (rest?.icon || rest?.shouldShowRightIcon) ? styles.alignItemsStretch : undefined,
```

Comment:
this is just for type narrowing , there are two ways of using Button component one with children like 
```
<Button>Hello</Button
```
or with text props like 
```
<Button text="Hello" />
```
and those two props `icon` and `shouldShowRightIcon` can be used only when `Button` is used with `text` prop. `'text' in rest` will assure that `Button` is used with `text` prop

Reply:
about the `?` its needed only when we are using object dot notation so if `rest` is undefined it will be safe to use properties of the object. when using `'text' in rest` and rest is `undefined` it will just return `false`